contracts/application/versioning/ImplementationProvider.sol

```diff
@@ -2,13 +2,13 @@ pragma solidity ^0.4.21;
 
 /**
  * @title ImplementationProvider
- * @dev Interface for providing implementation addresses for other contracts by name
+ * @dev Interface for providing implementation addresses for other contracts by name.
  */
 interface ImplementationProvider {
   /**
-   * @dev Prototype of function that returns implementation addresses
-   * @param contractName Name of the contract whose address is being fetched
-   * @return Implementation address of desired contract
+   * @dev Prototype of function that returns the implementation address of a contract.
```

Comment:
I like "prototype", but we used "abstract" in `BaseApp`, make consistent?

Reply:
I don't like prototype, good call to use abstract.